homeassistant/components/arwn/sensor.py

```diff
@@ -118,13 +114,28 @@ def __init__(self, name, state_key, units, icon=None):
         self.event = {}
         self._unit_of_measurement = units
         self._icon = icon
+        self._topic = topic
 
     def set_event(self, event):
         """Update the sensor with the most recent event."""
         self.event = {}
         self.event.update(event)
         self.async_write_ha_state()
 
+    async def async_added_to_hass(self):
+        """Entity has been added to hass."""
+
+        @callback
+        def message_received(msg):
+            """On message received on entity's topic."""
+            event = json.loads(msg.payload)
```

Comment:
The topic and payload parsing etc should be handled by a 3rd party library.

https://developers.home-assistant.io/docs/creating_component_code_review#4-communication-with-devicesservices